pcg/src/main/kotlin/nl/perfana/config/generator/PerfanaConfigGeneratorController.kt

```diff
@@ -46,11 +35,29 @@ class PerfanaConfigGeneratorController(val storage: FileStorage) {
             workDir,
             "source /root/.bashrc && /files/generate-config.sh $zipFileName $generatedFilesDir"
         )
+        val perfana_client = readPerfanaClientValue(workDir)
+
+        // create the projectId
+        val projectId = "perfana-starter-${perfana_client}-${timeStamp}"
 
         // now zip generated files
         val zipFile = storage.zip(workDir.resolve(generatedFilesDir), workDir, projectId)
         // keep in mapping for future download
         zipFiles[projectId] = zipFile
+
+        return projectId
+    }
+
+    private fun readPerfanaClientValue(workDir: Path): String {
+        // get the value of the PERFANA_CLIENT variable from setup.sh
+        val shellCommand = listOf("/bin/bash", "-c", "source ${workDir}/unzip/setup.sh && echo \$PERFANA_CLIENT")
+        val process = ProcessBuilder(shellCommand).start()
+        val reader = InputStreamReader(process.inputStream)
+        val perfana_client = reader.readText().trim()
+        println("Result: $perfana_client")
+        reader.close()
```

Comment:
put close and destroy in finally to avoid resource leaks at exceptions

Reply:
I know this code is not failsafe. Same goes for `private fun executeCommand(workDir: Path, command: String)`
Best way to solve it is to not rely on shell commands.